src/index/LocatedTriples.cpp

```diff
@@ -0,0 +1,243 @@
+                const CompressedBlockMetadata::PermutedTriple& triple) {
+              return block < triple;
+            },
+            [](const CompressedBlockMetadata& block) {
+              return block.lastTriple_;
+            }) -
+        blockMetadata.begin();
+    out.emplace_back(currentBlockIndex, triple, shouldExist);
+    cancellationHandle->throwIfCancelled();
+  }
+
+  return out;
+}
+
+// ____________________________________________________________________________
+bool LocatedTriplesPerBlock::hasUpdates(size_t blockIndex) const {
+  return map_.contains(blockIndex);
+}
+
+// ____________________________________________________________________________
+NumAddedAndDeleted LocatedTriplesPerBlock::numTriples(size_t blockIndex) const {
+  // If no located triples for `blockIndex_` exist, there is no entry in `map_`.
+  if (!hasUpdates(blockIndex)) {
+    return {0, 0};
+  }
+
+  auto blockUpdateTriples = map_.at(blockIndex);
+  size_t countInserts = std::ranges::count_if(
+      blockUpdateTriples, &LocatedTriple::shouldTripleExist_);
+  return {countInserts, blockUpdateTriples.size() - countInserts};
+}
+
+// ____________________________________________________________________________
+// Collect the relevant entries of a LocatedTriple into a triple.
+template <size_t numIndexColumns>
+auto tieIdTableRow(auto& row) {
+  if constexpr (numIndexColumns == 3) {
+    return std::tie(row[0], row[1], row[2]);
+  } else if constexpr (numIndexColumns == 2) {
+    return std::tie(row[0], row[1]);
+  } else {
+    return std::tie(row[0]);
+  }
+}
+
+// ____________________________________________________________________________
+// Collect the relevant entries of a LocatedTriple into a triple.
+template <size_t numIndexColumns>
+auto tieLocatedTriple(auto lt) {
+  if constexpr (numIndexColumns == 3) {
+    return std::tie(lt->triple_.ids_[0], lt->triple_.ids_[1],
+                    lt->triple_.ids_[2]);
+  } else if constexpr (numIndexColumns == 2) {
+    return std::tie(lt->triple_.ids_[1], lt->triple_.ids_[2]);
+  } else {
+    return std::tie(lt->triple_.ids_[2]);
+  }
+}
+
+// ____________________________________________________________________________
+template <size_t numIndexColumns>
+IdTable LocatedTriplesPerBlock::mergeTriplesImpl(size_t blockIndex,
+                                                 const IdTable& block) const {
+  // This method should only be called if there are located triples in the
+  // specified block.
+  AD_CONTRACT_CHECK(map_.contains(blockIndex));
+
+  AD_CONTRACT_CHECK(numIndexColumns <= block.numColumns());
+  AD_CONTRACT_CHECK(block.numColumns() >= 1);
+
+  auto numInsertsAndDeletes = numTriples(blockIndex);
+  IdTable result{block.numColumns(), block.getAllocator()};
+  result.resize(block.numRows() + numInsertsAndDeletes.numAdded_);
+
+  const auto& locatedTriples = map_.at(blockIndex);
+
+  auto cmpLt = [](auto lt, auto row) {
+    return tieIdTableRow<numIndexColumns>(row) >
```

Comment:
Maybe take the arguments as `const auto&` and turn the order arount in the implementation s.t. it actually reads `<` for less than.